<?xml version="1.0"?>

<launch>
    <!-- Set parameters -->
    <arg name="assignment_number"  default="1"/>
    <arg name="areas_filename"  default="areas_lab.yaml"/>

    <!-- Set ROS log -->
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find main_controller)/launch/rosconsole.conf"/>
    <env name="ROSCONSOLE_FORMAT" value="[${severity} ] [${node}]: ${message}"/>

    <!-- World configuration -->
    <rosparam command="load" file="$(find main_controller)/launch/$(arg areas_filename)" />

    <!-- YOLO -->
    <!-- <include file="$(find robotic_vision)/launch/yolov5.launch" /> -->
    <include file="$(find robotic_vision)/launch/shelfino_only.launch" />

    <!-- C++ code (controllers) -->
    <node pkg="ur5_controller" type="ur5_controller_node" name="ur5_controller_node" output="screen" />
    <node pkg="shelfino_controller" type="shelfino_controller_node" name="shelfino_controller_node" output="screen" />

    <!-- Main controller -->
    <node pkg="main_controller" type="fsm" name="fsm" output="screen">
        <param name="assignment"    value="$(arg assignment_number)"/>
    </node>

</launch>
